### ISSUE TITLE
Success Rate image in paper？

### ISSUE THREAD

**Issue by Folly135** [namsan96/SiMPL#4]
Hi! I tried to reproduce your experiment, but I didn't find the success rate images in the paper in the results. Were these images generated in the code? Or does it require additional work to generate?

Thank you very much.

**Comment by namsan96** [comment]
Hi, thank you for your interest on our work!

Our reproducing code doesn't directly include the plotting code for generating success rate plots.
But we have plotted the data field '[tr_return](https://github.com/namsan96/SiMPL/blob/f5286924bd0502431b1f9904db631f6b1e13dce7/reproduce/simpl_fine_tune.py#L22)' logged by current code.

**Comment by Folly135** [comment]
Thank you for your reply!
Does this mean that the success rate is the average of 10 Fine Tune rewards?
And I have some other question,
1. How do I change Maze env's seed? Just change **env = gym.make('simpl-maze-size20-seed0-v0')** into env = **gym.make('simpl-maze-size20-seed1-v0')** or **env = gym.make('simpl-maze-size20-seed2-v0')** ？
2. Can't the Kitchen env change the seed?
3. I don't quite understand the settings for **top** and **bottom** tasks in Maze Env. I guess it can be reach by changing the **train_tasks** or **tasks** in **train_tasks = Size20Seed0Tasks.flat_40t_train_tasks** or **tasks = Size20Seed0Tasks.flat_test_tasks** . How should this part be modified to achieve the effect of **Figure 6**?

Thank you very much!

**Comment by namsan96** [comment]
Yes, average of fine-tuning reward over [10 different tasks](https://github.com/namsan96/SiMPL/blob/f5286924bd0502431b1f9904db631f6b1e13dce7/reproduce/simpl_fine_tune.py#L86), and this is repeated for three times and finally averaged for the plot in the figure.

1.  You should register a new env to gym as in [here](https://github.com/namsan96/SiMPL/blob/f5286924bd0502431b1f9904db631f6b1e13dce7/simpl/env/maze/register.py)
2. We didn't consider any randomness in Kitchen environment (ex. object initial positions & states & robot initial pose) as it is in provided environment in D4RL. You might have to modify [D4RL code ](https://github.com/kpertsch/d4rl/blob/a4e37c6a9469a7dbf8849496e0ba4f912f921cb3/d4rl/kitchen/adept_envs/franka/kitchen_multitask_v0.py#L27)to introduce randomness on the Kitchen environment.
3. Yes, it can be achieved by setting different training and test task set. I will update the task set in Figure 13 (which was used for Figure 6) soon in this repository.

**Comment by namsan96** [comment]
The task set for train/test task alignment has been added to our code base (afc6c806523044211d072ad188e767a5f61c76f1).

You can use

```
train_tasks = Size20Seed0Tasks.narrow_train_goals  # in simpl_meta_train_10t.py

tasks = Size20Seed0Tasks.narrow_top_test_goals  # in simpl_fine_tune.py
```

to achieve the effect of Figure 6(b), and
```
train_tasks = Size20Seed0Tasks.narrow_train_goals  # in simpl_meta_train_10t.py

tasks = Size20Seed0Tasks.narrow_bottom_test_goals  # in simpl_fine_tune.py
```
to achieve the effect of Figure 6(c).

**Comment by Folly135** [comment]
Thank you very much for your reply!
I have some new questions about your previous replay. I used to think that **repeated for three times** means I should use three different seeds (seed0, seed1, seed2), but I found that the experimental results can also vary greatly when using the same seed in Maze_20t.
I guess this is because two different stages, making buffer and sampling. Is this normal?
And the **repeated for three times** means using only one seeds?

Thank you very much!

**Comment by Folly135** [comment]
I'm sorry to bother you, but I have a bit of uncertainty about the Maze Environment part you mentioned earlier.
**Yes, average of fine-tuning reward over 10 different tasks, and this is repeated for three times and finally averaged for the plot in the figure.**
I'm a bit confused about whether this part uses register to create seed1 and seed2, or does it run seed0 three times?
My understanding is that three different seeds should be used as mentioned in the paper. (seed0, seed1 and seed2)
Is my understanding correct?

I hope to receive your reply.
Thank you very much!

**Comment by namsan96** [comment]
Hi, we used the same maze layout for all 20x20 maze experiments.

If you want to meta-train on any different maze layout, you should use appropriate version of a pre-trained SPiRL policy.
It would be not possible to train a maze layout independent SPiRL with state-based observation, i.e. (x,y,vx,vy), as it doesn't give any information about the maze layout.
In this case SPiRL should be trained for each maze layout you want to train your agent.
In our experiment we used one maze layout along with a SPiRL policy that is trained on a dataset collected from the same layout.

Another option is to consider agent-centric image observation space so that one SPiRL can be generalized over many different maze layouts.

**Comment by hang925** [comment]
Hi! I tried to reproduce your experiment, but i donot know how to reproduce Figure 4,  especially the image of MTRL method, What additional work do I need to do? 
Thank you very much
